### PR TITLE
Add support for multi-schema migrations in Postgres

### PR DESCRIPTION
**Pull request by vporoshok**
I've save interface of function `GenerateAdvisoryLockId` with variadic params. So I've just add test to ensure that lock id depend on this additional parameters.

### CODE REVIEW

File: database/util.go [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
 	"fmt"
 	"hash/crc32"
 )
 
 const advisoryLockIdSalt uint = 1486364155
 
 // GenerateAdvisoryLockId inspired by rails migrations, see https://goo.gl/8o9bCT
-func GenerateAdvisoryLockId(databaseName string) (string, error) {
-	sum := crc32.ChecksumIEEE([]byte(databaseName))
+func GenerateAdvisoryLockId(databaseName string, additionalNames ...string) (string, error) {
```
Comment (dhui) [inline database/util.go]: I like the usage of variadic args!

File: database/postgres/postgres.go [hunk 2]
```diff
@@ -35,6 +35,7 @@ var (
 type Config struct {
 	MigrationsTable string
 	DatabaseName    string
+	SchemaName      string
```
Comment (dhui) [inline database/postgres/postgres.go]: I'm not a fan of the schema name being exported, but you're just following the existing pattern where the database name is exported. Let's leave this as is for now and we'll tackle unnecessarily exported field names in a later backwards incompatible version of `migrate`.

File: database/util.go (hunk 1 above)
Comment (vporoshok) [reply database/util.go]: Thank you!

File: database/util.go [hunk 3]
```diff
@@ -1,15 +1,21 @@
...
+	buf := bytes.NewBufferString(databaseName)
```
[the remark this answers is not in the document]
Comment (vporoshok) [reply database/util.go]: Wow! It's a good tip! Thank you!

**Comment by vporoshok** [comment]
Thank you!

**Comment by dhui** [comment]
Thanks for reporting and fixing the issue!

**Pull request merged**